### ISSUE TITLE
strange pronunciation on sum strings

### ISSUE THREAD

**Issue by alex19EP** [espeak-ng/espeak-ng#1533]
I think it's another ssml bug.

when speech-dispatcher read's  string: any package in our repository (by reading BUILDINFO) and then cleaned up. [1]
number 1 is read incorrectly.

you can reproduce this with command:
```bash
espeak-ng -v ru+max -s 449 -m '<speak><mark name="0:3"/>any <mark name="4:11"/>package <mark name="12:14"/>in <mark name="15:18"/>our <mark name="19:29"/>repository <mark name="30:33"/>(by <mark name="34:41"/>reading <mark name="42:52"/>BUILDINFO) <mark name="53:56"/>and <mark name="57:61"/>then <mark name="62:69"/>cleaned <mark name="70:73"/>up.<mark name="__spd_0"/> <mark name="74:77"/> 1</speak>'
```

**Comment by djphoenix** [comment]
It seems that these two are deep in a core. Maybe dictionaries are affect it too…

**Comment by djphoenix** [comment]
@jaacoppi @alex19EP can your cases be somehow reproduced out of SSML?

**Comment by alex19EP** [comment]
> can your cases be somehow reproduced out of SSML?

I think no.

**Comment by djphoenix** [comment]
Seems like a root cause in this block: https://github.com/espeak-ng/espeak-ng/blob/master/src/libespeak-ng/readclause.c#L898-L903

But I can't understand a path of there variables and how it works later...

Also, primarily there is rule about changing `V#` to `ja` that is not seems correct for me: https://github.com/espeak-ng/espeak-ng/blob/master/phsource/ph_russian#L442-L453

And pronunciation of russian `1` was changed in this commit: https://github.com/espeak-ng/espeak-ng/commit/762eba359df9b7db5bda63a13c960b60ebc43fd2

Another point is that word `один` (that equals to number `1`) is translated to `Vd;'in_`, not `V#d;'in`, and so spelled correctly.

**Comment by jaacoppi** [comment]
> Seems like a root cause in this block: https://github.com/espeak-ng/espeak-ng/blob/master/src/libespeak-ng/readclause.c#L898-L903
> 
> But I can't understand a path of these variables and how it works later...

I've been trying to understand the connection between SSML and ReadClause() for years now. It's really confusing. All this string manipulation, embedded commands and so on makes my head hurt everytime I try.

**Comment by kirill-jjj** [comment]
@djphoenix, @jaacoppi, yes, i used V# bekoz V changes phoneme to a. V# not used anywhere else, except for my additions. But if anything, I didn't add the V# itself.
 if you want, you can remove the change of this phoneme to ja. About the problem of a non-working three-step reduction method, I wrote earlier
look, for the letter o we have this rule
        о       o
but о we have unstressed
so it changes it to V, but...
the next vowel is stressed, so it should change the phoneme to a
but this does not happen, because the first phoneme change has occurred, and espeak does not care if there are any other rules for changing phonemes, and then he justs exits
when I write a transcription in the list, whether I want it or not, I skip the first step, that is, the phoneme o
and it just replaces Vd;'in with ad;'in, because the next vowel is stressed.

**Comment by djphoenix** [comment]
@jaacoppi I was thinking at least three times to remove all of SSML code and rewrite it completely, based on e.g. libxml, and passing preprocessed string with embeddings to the synthesizer. Sometimes, start from scratch is better than keep code that no one can understand.

@kirill-jjj can you do something that fixes this issue? That definitely will be a crutch, as we can't fix this properly, but it will work. `adin` sounds better than `jadin` anyway.

**Comment by kirill-jjj** [comment]
@djphoenix, just change in ru_list
1    V#d;'in
to 1    ad;'in

**Comment by alex19EP** [comment]
@kirill-jjj I think removing if cases from v# phoneme would be a better option. can you do that?

**Comment by jaacoppi** [comment]
> @jaacoppi I was thinking at least three times to remove all of SSML code and rewrite it completely, based on e.g. libxml, and passing preprocessed string with embeddings to the synthesizer. Sometimes, start from scratch is better than keep code that no one can understand.

Sounds like a good idea. No need to try to parse xml ourselves when there's a better library available.
